### ISSUE TITLE
Restful.Login.checkCachedCredentials should get access token as `Maybe String`

### ISSUE THREAD

**Issue by amitaibu** [Gizra/elm-restful#9]
This will allow us to prevent calling the server is we have no access token

**Comment by rgrempel** [comment]
If you don't have an access token, you don't have to call `checkCachedCredentials` at all ... you can start with `loggedOut`.

**Comment by amitaibu** [comment]
Indeed, but since this will almost always be the case, we can instead of

```elm
        ( loginStatus, loginCmd ) =
            -- We kick off the process of checking the cached credentials which
            -- we were provided.
            case flags.accessToken of
                Just accessToken ->
                    Restful.Login.checkCachedCredentials
                        loginConfig
                        flags.backendUrl
                        accessToken

                Nothing ->
                    ( Restful.Login.AnonymousUser Nothing ()
                    , Cmd.none
                    )
```

do the "heavy lifting" and


```elm
          -- We kick off the process of checking the cached credentials which
          -- we were provided, or default to Logged out.
        ( loginStatus, loginCmd ) =
                    Restful.Login.checkCachedCredentials
                        loginConfig
                        flags.backendUrl
                        flags.accessToken -- <- This is a Maybe
```

**Comment by rgrempel** [comment]
Ah, yes, that does seem nicer ...

**Comment by rgrempel** [comment]
I've added this change to #11.

**Comment by rgrempel** [comment]
Merged in #11.
